### PR TITLE
chore(doc): update references of container registry of nfs-provisioner from gcr.io to k8s.gcr.io

### PR DESCRIPTION
**Pull request by mittachaitu**
This PR updates the container registry references of nfs-provisioner
 from `gcr.io/k8s-staging-sig-storage` to `k8s.gcr.io/sig-storage`

Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

### CODE REVIEW

**Comment by mittachaitu** [comment]
Do I need to update references in [cloudbuild.yaml](https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/blob/9dadbd8d051b7568f322bc9e75f2cb31c223f2ed/release-tools/cloudbuild.yaml#L35)?

**Comment by kvaps** [comment]
Hi, thanks for PR!
Cloud you also bump the chart version please?
https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/blob/9dadbd8d051b7568f322bc9e75f2cb31c223f2ed/deploy/helm/Chart.yaml#L5

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ all: build
 include release-tools/build.make
 
 ifeq ($(REGISTRY),)
-	REGISTRY = gcr.io/k8s-staging-sig-storage/
+	REGISTRY = k8s.gcr.io/sig-storage/
```
Comment (kmova) [inline Makefile]: This should not be changed.

**Comment by mittachaitu** [comment]
> Hi, thanks for PR!
> Cloud you also bump the chart version please?
> https://github.com/kubernetes-sigs/nfs-ganesha-server-and-external-provisioner/blob/9dadbd8d051b7568f322bc9e75f2cb31c223f2ed/deploy/helm/Chart.yaml#L5

Since this PR is all about image reference changes. Can I update it to `1.3.1`?

**Pull request merged**